website/docs/guides/container_service_extension_4_0.html.markdown

```diff
@@ -0,0 +1,410 @@
+- `github_personal_access_token`: Create this one [here](https://github.com/settings/tokens),
+  this will avoid installation errors caused by GitHub rate limiting, as the TKGm tenant creation process requires downloading
+  some Kubernetes components from GitHub.
+  The token should have the `public_repo` scope for classic tokens and `Public Repositories` for fine-grained tokens.
+- `cse_admin_user`: This should reference the CSE Administrator [User][user] that was created in Step 1.
+- `cse_admin_api_token`: This should be the API token that you created for the CSE Administrator after Step 1.
+
+### Final considerations
+
+### Verifying that the setup works
+
+To validate that the CSE Server is working correctly, you can either do it programmatically with a [SNAT rules][nat_rule] that maps
+one available IP to the CSE Server, or using the UI:
+
+- With a [SNAT rules][nat_rule] you could connect to the CSE Server VM through `ssh` and the credentials that are stores in the `terraform.tfstate` file,
+  with a resource similar to this:
+```
+resource "vcd_nsxt_nat_rule" "solutions_nat" {
+  org             = vcd_org.solutions_organization.name
+  edge_gateway_id = vcd_nsxt_edgegateway.solutions_edgegateway.id
+
+  name        = "CSE Server SNAT rule"
+  rule_type   = "SNAT"
+  description = "CSE Server SNAT rule"
+
+  external_address = "One available IP from Solutions Provider Gateway"
+  internal_address = "CSE Server IP"
+  logging          = true
+}
+```
+
+- With the UI, you can go to the CSE Server VM and open a **web console**. The credentials to login are shown in the _Guest customization properties Edit view_.
+
+Once you gain access to the CSE Server, you can check the `cse.log` file, the configuration file or check Internet connectivity.
+If something does not work, please check the **Troubleshooting** section below.
+
+#### Install the UI plugin
+
+To manage CSE clusters with the UI, you can [download the Kubernetes Container Clusters UI plug-in 4.0][cse_docs]
+and install it in your VCD appliance. If the old CSE 3.x plugin is installed, you will need to remove it first. The plugin
+will allow tenants to create Kubernetes clusters with the UI wizard. Providers should still use the proposed Terraform configuration
+to perform updates on the CSE Server (see sections below).
+
+#### Troubleshooting
+
+To evaluate the correctness of the setup, you can check the _"Verifying that the setup works"_ section above.
+
+-> You can visit [the documentation][cse_docs] to learn how to monitor the logs and troubleshoot possible problems.
+
+The most common issues are:
+
+- Lack of Internet connectivity in CSE Server:
+  - Verify that the IPs specified in your Provider Gateways are correct.
+  - Verify that the IPs specified in your Edge Gateways are correct.
+  - Verify that your Firewall setup is not blocking outbound connectivity.
+  - Verify that the Routed network has the DNS correctly set and working.
+
+- OVA upload is taking too long:
```

Comment:
Is it worth mentioning that the upload _might_ fail and the user might need to restart the process?

Reply:
I've added the following:

> - If upload fails, or you need to replace the OVA, you can do a `terraform apply -replace=vcd_catalog_vapp_template.cse_ova`.

Reply:
That's a nice trick @adambarreiro ! Consider adding it to the resource doc page too:
https://registry.terraform.io/providers/vmware/vcd/latest/docs/resources/catalog_vapp_template

Reply:
Done!